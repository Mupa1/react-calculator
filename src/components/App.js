/* eslint-disable react/no-unused-state */
import React from 'react';

import '../App.css';
import Display from './Display';
import ButtonPanel from './ButtonPanel';
import calculate from '../logic/calculate';

class App extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      total: null,
      next: null,
      operation: null,
    };

    this.handleClick = this.handleClick.bind(this);
  }

  handleClick = buttonName => {
    const result = calculate(this.state, buttonName);
    this.setState(result);
  }

  render() {
    const { total, next } = this.state;
    const result = next || total || '0';

    return (
      <>
        <Display result={result} />
        <ButtonPanel clickHandler={this.handleClick} />
      </>
    );
  }
}

export default App;
