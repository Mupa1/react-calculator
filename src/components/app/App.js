import React from 'react';
import { BrowserRouter, Switch, Route } from 'react-router-dom';

import Nav from '../nav/Nav';
import Home from '../home/Home';
import Quote from '../quote/Quote';
import Calculator from '../calculator/Calculator';

const App = () => (
  <>
    <BrowserRouter>
      <Nav />
      <Switch>
        <Route exact path="/" component={Home} />
        <Route path="/calculator" component={Calculator} />
        <Route path="/quote" component={Quote} />
      </Switch>
    </BrowserRouter>
  </>
);

export default App;
