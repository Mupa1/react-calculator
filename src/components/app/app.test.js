import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';

import App from './App';

it('renders correctly', () => {
  const tree = renderer
    .create(<App />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

describe('App', () => {
  test('renders App component', () => {
    render(<App />);
    expect(screen.getByText('Welcome to our page!')).toBeInTheDocument();
  });
});
