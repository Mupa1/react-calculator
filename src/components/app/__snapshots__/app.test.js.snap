// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <nav>
    <h1>
      Math Magicians
    </h1>
    <ul
      className="navLinks"
    >
      <a
        className="navLink"
        href="/"
        onClick={[Function]}
      >
        <li>
          Home
        </li>
      </a>
      <a
        className="navLink"
        href="/calculator"
        onClick={[Function]}
      >
        <li>
          Calculator
        </li>
      </a>
      <a
        className="navLink"
        href="/quote"
        onClick={[Function]}
      >
        <li>
          Quote
        </li>
      </a>
    </ul>
  </nav>,
  <div
    className="home"
  >
    <h2>
      Welcome to our page!
    </h2>
    <p>
      Math Magicians lets you do fun basic calculations.
    </p>
    <p>
      Click on the Calculator link to do some math.
    </p>
  </div>,
]
`;
